src/main/java/org/spongepowered/asm/mixin/transformer/InnerClassGenerator.java

```diff
@@ -421,7 +427,8 @@ private static String getUniqueReference(String originalName, ClassInfo targetCl
         if (name.matches("^[0-9]+$")) {
             name = "Anonymous";
         }
-        return String.format("%s$%s$%s", targetClass, name, UUID.randomUUID().toString().replace("-", ""));
+        UUID uuid = UUID.nameUUIDFromBytes(originalName.getBytes(StandardCharsets.UTF_8));
```

Comment:
The name of inner class can be duplicated. Isn't it?
How about pass the coordinate to this method and use the hex hash of the string?